packages/text-area/src/TextArea.tsx

```diff
@@ -167,11 +167,7 @@ export default function TextArea({
     }
   };
 
-  if (!label && !ariaLabelledBy) {
-    console.error(
-      'For screen-reader accessibility, label or aria-labelledby must be provided to IconButton.',
-    );
-  }
+  validateLabelProps({ label, ...rest }, 'TextArea');
```

Comment:
This function isn't equivalent to the previous check. We only check that `aria-label` or `aria-labelledby` is present with `validateLabelProps`. It will not check whether `label` exists.

This might have to look like:
```js
if (!label) {
  validateLabelProps(rest, 'TextArea');
}
```